### PR TITLE
mount: fix k8s pvc and os mount directory permission bug

### PR DESCRIPTION
**Pull request by hilimd**
**背景：**

我通过k8s pvc挂载卷启动的mysql/rabbitmq容器均出现error，查看日志提示数据目录找不到，通过排查原因为数据目录属组及权限不可修改

**复现：**
挂载命令：
```
weed mount -filer=10.0.100.51:30888 -dir=/tmp/limd  -filer.path=/buckets/limd -collection=limd -dirAutoCreate=true

chown -R mysql:mysql /tmp/limd
chown: /tmp/limd/: No such file or directory

```

问题原因是dir.entry为空导致，已做修改，k8s挂卷完美运行，请作者参考

另外修复了挂载目录属组及权限显示的问题，目前和NFS效果一样

### CODE REVIEW

File: weed/command/mount.go [hunk 1]
```diff
@@ -44,7 +44,7 @@ func init() {
 	mountOptions.cacheSizeMB = cmdMount.Flag.Int64("cacheCapacityMB", 1000, "local file chunk cache capacity in MB (0 will disable cache)")
 	mountOptions.dataCenter = cmdMount.Flag.String("dataCenter", "", "prefer to write to the data center")
 	mountOptions.allowOthers = cmdMount.Flag.Bool("allowOthers", true, "allows other users to access the file system")
-	mountOptions.umaskString = cmdMount.Flag.String("umask", "022", "octal umask, e.g., 022, 0111")
+	mountOptions.umaskString = cmdMount.Flag.String("umask", "000", "octal umask, e.g., 022, 0111")
```
Comment (chrislusf) [inline weed/command/mount.go]: no need to change the default value

File: weed/s3api/s3api_object_handlers.go [hunk 2]
```diff
@@ -112,6 +112,12 @@ func (s3a *S3ApiServer) DeleteObjectHandler(w http.ResponseWriter, r *http.Reque
 
 	bucket, object := getBucketAndObject(r)
 
+	response, _ := s3a.listFilerEntries(bucket, object, 1, "", "/")
```
Comment (chrislusf) [inline weed/s3api/s3api_object_handlers.go]: not related to this issue

File: weed/s3api/s3api_object_handlers.go [hunk 3]
```diff
@@ -178,6 +183,11 @@ func (s3a *S3ApiServer) DeleteMultipleObjectsHandler(w http.ResponseWriter, r *h
 	s3a.WithFilerClient(func(client filer_pb.SeaweedFilerClient) error {
 
 		for _, object := range deleteObjects.Objects {
+			response, _ := s3a.listFilerEntries(bucket, object.ObjectName, 1, "", "/")
```
Comment (chrislusf) [inline weed/s3api/s3api_object_handlers.go]: not related to this issue

File: weed/filesys/dir.go [hunk 4]
```diff
@@ -82,9 +82,9 @@ func (dir *Dir) Getxattr(ctx context.Context, req *fuse.GetxattrRequest, resp *f
 func (dir *Dir) setRootDirAttributes(attr *fuse.Attr) {
 	attr.Inode = 1 // filer2.FullPath(dir.Path).AsInode()
 	attr.Valid = time.Hour
-	attr.Uid = dir.wfs.option.MountUid
-	attr.Gid = dir.wfs.option.MountGid
-	attr.Mode = dir.wfs.option.MountMode
+	attr.Uid = dir.entry.Attributes.Uid
```
Comment (chrislusf) [inline weed/filesys/dir.go]: where does the value for `dir.entry.Attributes.Uid` come from?

Comment (hilimd) [reply weed/filesys/dir.go]: 新目录时Uid默认为0，所以不需要在初始时设置，weed/pb/filer_pb.go，792行

```
func (x *FuseAttributes) GetUid() uint32 {
	if x != nil {
		return x.Uid
	}
	return 0
}
```

File: weed/command/mount.go (hunk 1 above)
Comment (hilimd) [reply weed/command/mount.go]: 已恢复

File: weed/s3api/s3api_object_handlers.go (hunk 2 above)
Comment (hilimd) [reply weed/s3api/s3api_object_handlers.go]: 已恢复

File: weed/s3api/s3api_object_handlers.go (hunk 3 above)
Comment (hilimd) [reply weed/s3api/s3api_object_handlers.go]: 已恢复

File: weed/filesys/dir.go (hunk 4 above)
Comment (chrislusf) [reply weed/filesys/dir.go]: can you use `-map.uid` and `-map.gid`to control instead?

Comment (chrislusf) [reply weed/filesys/dir.go]: Currently, the root uid/gid is the same as the mounted directory.

With this change, the uid/gid will all be zero.

Comment (hilimd) [reply weed/filesys/dir.go]: 不会的，会通过filer_pb.GetEntry获取到正确的uid和gid

**Pull request closed without merging**